src/python/pants/engine/target.py

```diff
@@ -1705,12 +1720,25 @@ def compute_value(
 class StringSequenceField(SequenceField[str]):
     expected_element_type = str
     expected_type_description = "an iterable of strings (e.g. a list of strings)"
+    valid_choices: ClassVar[Optional[Union[Type[Enum], Tuple[str, ...]]]] = None
```

Comment:
My original implementation didn't account for Enums like StringField does, so I copied the valid_choices definition from StringField.